Skyline.DataMiner.CICD.Tools.Validator/ValidatorRunner.cs

```diff
@@ -0,0 +1,230 @@
+﻿namespace Skyline.DataMiner.CICD.Tools.Validator
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    using Microsoft.Build.Locator;
+    using Microsoft.Extensions.Logging;
+
+    using Skyline.DataMiner.CICD.Parsers.Common.VisualStudio;
+    using Skyline.DataMiner.CICD.Parsers.Common.VisualStudio.Projects;
+    using Skyline.DataMiner.CICD.Tools.Reporter;
+    using Skyline.DataMiner.CICD.Tools.Validator.OutputWriters;
+
+    internal class ValidatorRunner
+    {
+        private readonly ILogger logger;
+
+        public ValidatorRunner()
+        {
+            using (ILoggerFactory factory = LoggerFactory.Create(builder => builder.AddSimpleConsole(options => { options.IncludeScopes = true; options.SingleLine = true; })))
+            {
+                logger = factory.CreateLogger("Validator");
+            }
+        }
+
+        /// <summary>
+        /// Validates the specified solution.
+        /// </summary>
+        /// <param name="solutionFilePath">The solution (.sln) file path.</param>
+        /// <param name="validatorResultsOutputDirectory">The output directory.</param>
+        /// <param name="validatorResultsFileName">The file name of the results.</param>
+        /// <param name="outputFormats">The output formats.</param>
+        /// <param name="includeSuppressed"><c>true</c> to include suppressed results; otherwise, <c>false</c>.</param>
+        /// <param name="performBuild"><c>true</c> to perform a build; otherwise, <c>false</c>.</param>
+        /// <param name="buildTimeout">The build timeout.</param>
+        /// <returns>Result value 0 indicates success.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="solutionFilePath"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException">Invalid solution file path.</exception>
+        public async Task<int> ValidateProtocolSolution(string solutionFilePath, string validatorResultsOutputDirectory, string validatorResultsFileName, string[] outputFormats, bool includeSuppressed, bool performBuild, int buildTimeout)
+        {
+            if (solutionFilePath == null) throw new ArgumentNullException(nameof(solutionFilePath));
+
+            if (String.IsNullOrEmpty(solutionFilePath)) throw new ArgumentException("Invalid solution file path.", nameof(solutionFilePath));
+
+            solutionFilePath = Path.GetFullPath(solutionFilePath);
+
+            if (!File.Exists(solutionFilePath)) throw new ArgumentException($"The specified solution file '{solutionFilePath}' does not exist.", nameof(solutionFilePath));
+
+            // Required for both building solution and for loading the MSBuildWorkspace to perform validation.
+            if (!MSBuildLocator.IsRegistered)
+            {
+                MSBuildLocator.RegisterDefaults();
+            }
```

Comment:
Maybe move this to just before the 'if (performBuild)'. This way this isn't executed when it's a legacy style solution and/or not .NET Framework 4.8